tests/test_create_pod_attaching_multiple_pvc_start_IO.py

```diff
@@ -0,0 +1,376 @@
+import os
+import yaml
+import logging
+import base64
+import pytest
+import ocs.ocp
+import ocs.pod
+import ocs.defaults as defaults
+
+from ocs import exceptions
+from time import sleep
+from ocsci import tier1, ManageTest
+from ocsci.enums import StatusOfTest
+from utility import templating
+
+log = logging.getLogger(__name__)
+
+CBP = ocs.ocp.OCP(
+    kind='CephBlockPool', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+SEC = ocs.ocp.OCP(
+    kind='Secret', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+SC = ocs.ocp.OCP(
+    kind='StorageClass', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+PVC = ocs.ocp.OCP(
+    kind='PersistentVolumeClaim', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+POD = ocs.ocp.OCP(
+    kind='Pod', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+OCP = ocs.ocp.OCP(
+    kind='Service', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+DC = ocs.ocp.OCP(
+    kind='DeploymentConfig', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+
+TEMP_YAML = os.path.join("templates/ocs-deployment", "temp.yaml")
+TEMPLATES_DIR = "templates/CSI/rbd"
+CIRROS_DIR = "templates/app-pod-yamls"
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Removing the resources created for this test
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Setting up the environment for the test
+    """
+    create_ceph_block_pool()
+    create_rbd_secret()
+    create_rbd_storage_class()
+
+
+def teardown(self):
+    """
+    Teardown the created Environment
+    """
+    log.info(f"Cirros pod name {self.cirros_name[0]}")
+    delete_pod(self.cirros_name[0])
+    delete_persistent_volume(self.pv_list)
+    delete_storage_class()
+    delete_pool()
+    delete_secret()
+    os.remove(TEMP_YAML)
+
+
+def create_ceph_block_pool():
+    """
+    Create ceph block pool
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "CephBlockPool.yaml")
+    log.info(f"Create a ceph block pool")
+    data = templating.generate_yaml_from_jinja2_template_with_data(
+        template
+    )
+    ceph_block_pool = data['metadata']['name']
+    with open(TEMP_YAML, 'w') as yaml_file:
+        yaml.dump(data, yaml_file)
+    assert CBP.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    sleep(5)
+    # Check for Pool creation success from ceph and OC
+    try:
+        CBP.get(ceph_block_pool)
+        log.info(f"Ceph Block Pool {ceph_block_pool} creation success")
+    except exceptions.CommandFailed:
+        assert log.info(
+            f'Exception BlockPool {ceph_block_pool} does not exists '
+        )
+        return False
+
+
+def create_rbd_secret():
+    """
+    Get rbd secret from ceph and create secret in oc
+
+    """
+    secret_data = {}
+    cmd_out = ocs.ocp.exec_ceph_cmd(ceph_cmd="ceph auth get-key client.admin")
+    key = cmd_out.get('key')
+    admin_key_byte = base64.b64encode(key.encode("utf-8"))
```

Comment:
please create a function for create_secret_key()

Reply:
ack, Will add this in library in ocp.py

Reply:
Created pr #136 for base64 encoded funciton